pkg/agent/flowexporter/flowrecords/flow_records.go

```diff
@@ -48,18 +56,19 @@ func (fr *FlowRecords) AddOrUpdateFlowRecord(key flowexporter.ConnectionKey, con
 			isIPv6 = true
 		}
 		record = flowexporter.FlowRecord{
-			Conn:               conn,
+			Conn:               *conn,
 			PrevPackets:        0,
 			PrevBytes:          0,
 			PrevReversePackets: 0,
 			PrevReverseBytes:   0,
+			TCPState:           conn.TCPState,
```

Comment:
this is not needed. The record "owns" the connection. We have to make a copy to avoid races: this is the comment I asked you to add for the function earlier.